BuildingBlocks/src/BuildingBlocks.Infrastructure/Persistence/Database/SaveChangesInterceptor.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Enmeshed.Tooling.Extensions;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Enmeshed.BuildingBlocks.Infrastructure.Persistence.Database;
+
+public class SaveChangesTimeInterceptor : SaveChangesInterceptor
```

Comment:
I wonder if this class is thread safe. So: is there a separate stopwatch for each save operation?

In other words: does EF Core create a new object of this class every time SaveChangesAsync is called?

Reply:
It is actually thread-safe. I've implemented the static counter which is incremented in the constructor whenever an interceptor is instantiated. The value increases for every request made.